api/sandbox.go

```diff
@@ -0,0 +1,114 @@
+package api
+
+import (
+	"context"
+	"syscall"
+
+	"github.com/hyperhq/runv/api"
+	ocispecs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type ContainerConfig struct {
+	Id string
+
+	UGI    *UserGroupInfo
+	Mounts map[string]Mount
+
+	OciSpec ocispecs.Spec
+}
+
+type Mount struct {
+	StorageID string
+	Path      string
+	ReadOnly  bool
+}
+
+type UserGroupInfo struct {
+	User             string
+	Group            string
+	AdditionalGroups []string
+}
+
+type ProcessConfig struct {
+	Container string
+	Id        string
+	UGI       *UserGroupInfo
+
+	OciProcess ocispecs.Process
+}
+
+type Sandbox struct {
+	VM    VM
+	Agent Agent
+
+	Devices map[string]AddedDevice
+
+	Containers map[string]Container
+}
+
```

Comment:
virtcontainers does not export the `Pod` structure directly to the top level API, but the API caller should use the `PodConfig` structure instead:

```Go
// PodConfig is a Pod configuration.
type PodConfig struct {
	ID string

	Hostname string

	// Field specific to OCI specs, needed to setup all the hooks
	Hooks Hooks

	// VMConfig is the VM configuration to set for this pod.
	VMConfig Resources

	HypervisorType   HypervisorType
	HypervisorConfig HypervisorConfig

	AgentType   AgentType
	AgentConfig interface{}

	ProxyType   ProxyType
	ProxyConfig ProxyConfig

	ShimType   ShimType
	ShimConfig interface{}

	NetworkModel  NetworkModel
	NetworkConfig NetworkConfig

	// Volumes is a list of shared volumes between the host and the Pod.
	Volumes []Volume

	// Containers describe the list of containers within a Pod.
	// This list can be empty and populated by adding containers
	// to the Pod a posteriori.
	Containers []ContainerConfig

	// Annotations keys must be unique strings and must be name-spaced
	// with e.g. reverse domain notation (org.clearlinux.key).
	Annotations map[string]string
}
```